src/control/cmd/agent/infocache.go

```diff
@@ -0,0 +1,139 @@
+//
+// (C) Copyright 2020 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package main
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+
+	mgmtpb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/daos-stack/daos/src/control/drpc"
+	"github.com/daos-stack/daos/src/control/lib/atm"
+	"github.com/daos-stack/daos/src/control/lib/netdetect"
+	"github.com/daos-stack/daos/src/control/logging"
+)
+
+const defaultNumaNode = 0
+
+type attachInfoCache struct {
+	log logging.Logger
+	// is caching enabled?
+	enabled atm.Bool
+	// is the cache initialized?
+	initialized atm.Bool
+	// maps NUMA affinity and device index to a response
+	resmgmtpb map[int]map[int][]byte
+	// maps NUMA affinity to a device index
+	devIdx map[int]int
+	mutex  sync.Mutex
+}
+
+// loadBalance is a simple round-robin load balancing scheme
+// to assign network interface adapters to clients
+// on the same NUMA node that have multiple adapters
+// to choose from.  Returns the index of the device to use.
+func (aic *attachInfoCache) loadBalance(numaNode int) int {
+	aic.mutex.Lock()
+	numDevs := len(aic.resmgmtpb[numaNode])
+	// keep the deviceIndex in range of the current resmgmtpb map
+	deviceIndex := aic.devIdx[numaNode] % numDevs
```

Comment:
It shouldn't be required if everything is functioning.  I thought it was small price to pay for ensuring the index was sane.   I've removed it though, since you are right that it is not strictly required.